rskj-core/src/main/java/co/rsk/cli/tools/DbMigrate.java

```diff
@@ -103,32 +110,52 @@ protected void onExecute(@Nonnull String[] args, @Nonnull RskContext ctx) throws
             throw new IllegalArgumentException("Db to migrate not specified. Please specify in the first argument.");
         }
 
-        DbKind sourceDbKind = ctx.getRskSystemProperties().databaseKind();
+        DbKind defaultDbKind = ctx.getRskSystemProperties().databaseKind();
         DbKind targetDbKind = DbKind.ofName(args[0]);
 
-        if (sourceDbKind == targetDbKind) {
-            throw new IllegalArgumentException(String.format(
-                    "Cannot migrate to the same database, current db is %s and target db is %s",
-                    sourceDbKind,
-                    targetDbKind
-            ));
-        }
-
         String sourceDbDir = ctx.getRskSystemProperties().databaseDir();
         String targetDbDir = sourceDbDir + "_tmp";
 
-        logger.info("Preparing to migrate from {} to {}", sourceDbKind.name(), targetDbKind.name());
+        logger.info("Preparing to migrate to {}", targetDbKind.name());
 
         FileUtil.recursiveDelete(targetDbDir);
 
         try (Stream<Path> databasePaths = Files.list(Paths.get(sourceDbDir))) {
-            databasePaths.filter(path -> path.toFile().isDirectory())
-                    .map(path -> path.getFileName().toString())
-                    .map(dbName -> buildDbMigrationInformation(sourceDbDir, targetDbDir, sourceDbKind, targetDbKind, dbName))
-                    .forEach(this::migrate);
-        }
+            List<Path> paths = databasePaths.collect(Collectors.toList());
+
+            for (Path path : paths) {
+                if (!path.toFile().isDirectory()) {
+                    continue;
+                }
+
+                String fullSourceDbPath = path.toAbsolutePath().toString();
+                String dbName = path.getFileName().toString();
+                String fullTargetDbPath = String.format("%s/%s", targetDbDir, dbName);
+
+                DbKind dbSourceDbKind = KeyValueDataSource.getDbKindValueFromDbKindFile(fullSourceDbPath, defaultDbKind);
+
+                if (dbSourceDbKind == targetDbKind) {
+                    File targetDbFolder = new File(targetDbDir);
+                    targetDbFolder.mkdirs();
+                    Files.move(Paths.get(fullSourceDbPath), Paths.get(fullTargetDbPath));
+                    continue;
+                }
+
+                DbMigrationInformation dbMigrationInformation = buildDbMigrationInformation(sourceDbDir, targetDbDir, dbSourceDbKind, targetDbKind, dbName, ctx);
+
+                this.migrate(dbMigrationInformation);
 
-        KeyValueDataSource.validateDbKind(targetDbKind, targetDbDir, true);
+                KeyValueDataSource.generatedDbKindFile(targetDbKind, fullTargetDbPath);
+
+                String nodeIdFilePath = "/" + NODE_ID_FILE;
+
+                if (new File(fullSourceDbPath, NODE_ID_FILE).exists()) {
```

Comment:
it seems that there's no need to check for existence of `nodeId.properties` in subfolders, as this this can only be located in a db root folder.

Reply:
Addressed